src/test/java/org/sagebionetworks/bridge/exporter/helper/ExportHelperTest.java

```diff
@@ -61,4 +86,78 @@ public void uploadFreeformText() throws Exception {
         String attachmentText = new String(attachmentBytesCaptor.getValue(), Charsets.UTF_8);
         assertEquals(attachmentText, DUMMY_ATTACHMENT_CONTENT);
     }
+
+    @Test
+    public void getEndDateTimeTest() {
+        BridgeExporterRequest request;
+        DateTime endDateTime;
+        ExportHelper exportHelper = new ExportHelper();
+
+        // DAILY
+        request = new BridgeExporterRequest.Builder().withDate(UPLOAD_DATE_OBJ).build();
+        endDateTime = exportHelper.getEndDateTime(request);
+
+        assertEquals(endDateTime.getMillis(), UPLOAD_END_DATE_TIME_OBJ.getMillis());
+
+        // HOURLY
+        request = new BridgeExporterRequest.Builder().withStartDateTime(UPLOAD_START_DATE_TIME_OBJ)
+                .withEndDateTime(UPLOAD_END_DATE_TIME_OBJ)
+                .withStudyWhitelist(ImmutableSet.of("dummy-whitelist"))
+                .build();
+        endDateTime = exportHelper.getEndDateTime(request);
+
+        assertEquals(endDateTime.getMillis(), UPLOAD_END_DATE_TIME_OBJ.getMillis());
+
+        // s3 override
+        request = new BridgeExporterRequest.Builder().withRecordIdS3Override("dummy-override").build();
+        endDateTime = exportHelper.getEndDateTime(request);
+
+        assertNull(endDateTime);
+    }
+
+    @Test
+    public void bootstrapStudyIdsToQueryTest() {
+        // mock ddb client with scan result as last export date time
+        AmazonDynamoDBClient mockDdbClient = mock(AmazonDynamoDBClient.class);
+
+        // mock study table and study id list
+        Table mockStudyTable = mock(Table.class);
+        when(mockStudyTable.getTableName()).thenReturn(STUDY_TABLE_NAME);
+
+        List<Map<String, AttributeValue>> studyIdList = ImmutableList.of(
+                ImmutableMap.of(IDENTIFIER, new AttributeValue().withS("ddb-foo")),
+                ImmutableMap.of(IDENTIFIER, new AttributeValue().withS("ddb-bar"))
+        );
+
+        ScanResult scanResult = new ScanResult();
+        scanResult.setItems(studyIdList);
+        ScanRequest scanRequest = new ScanRequest().withTableName(STUDY_TABLE_NAME);
+        when(mockDdbClient.scan(eq(scanRequest))).thenReturn(scanResult);
+
+        ExportHelper exportHelper = new ExportHelper();
+        exportHelper.setDdbClientScan(mockDdbClient);
+        exportHelper.setDdbStudyTable(mockStudyTable);
+
+        // mock request
+        BridgeExporterRequest request;
+        List<String> studyIdsToUpdate;
+        // daily
+        request = new BridgeExporterRequest.Builder().withDate(UPLOAD_DATE_OBJ).build();
+        studyIdsToUpdate = exportHelper.bootstrapStudyIdsToQuery(request);
+
+        assertEquals(studyIdsToUpdate.size(), 2);
+        assertEquals(studyIdsToUpdate.get(0), studyIdList.get(0).get(IDENTIFIER).getS());
+        assertEquals(studyIdsToUpdate.get(1), studyIdList.get(1).get(IDENTIFIER).getS());
```

Comment:
This is a facetious test. If your code has a bug, so do your tests. You should instead test against what you actually expect the result to be.